frontend/src/utils/utils.ts

```diff
@@ -40,6 +40,15 @@ export function mapErrorCodeToMessage(code: string) {
     }
 }
 
+//stolen from https://stackoverflow.com/a/12646864/9124516
+//mutates original array and returns it
+export function durstenfeldShuffle(array: any[]) {
```

Comment:
I'm curious, why use different shuffle methods in getXPairs and the pairing game screen? Would it be possible to just choose one?

Reply:
It seems like the `shuffle` function is not actually like random (it is biased in some way that I don't remember). That is why we added this one. We can use either one, which would you prefer?

Reply:
durnstenfeld shuffle ftw